packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -893,16 +944,24 @@ class IOSDeviceLogReader extends DeviceLogReader {
   // Strip off the logging metadata (leave the category), or just echo the line.
   String _debuggerLineHandler(String line) => _debuggerLoggingRegex.firstMatch(line)?.group(1) ?? line;
 
+  /// Use both logs from `idevicesyslog` and `ios-deploy` when debugging from CI system
+  /// since sometimes `ios-deploy` does not return the device logs:
+  /// https://github.com/flutter/flutter/issues/121231
+  @visibleForTesting
+  bool get useBothLogDeviceReaders {
+    return _usingCISystem && _majorSdkVersion >= 16;
+  }
+
   void _listenToSysLog() {
     // syslog is not written on iOS 13+.
```

Comment:
Should we update this comment? On the other hand, I'm not sure what to update it to, as I'm confused as to what's going on...

Reply:
What's confusing? The changes I made or just the way the streams work? But yes, I can update the comment

Reply:
I'm confused why idevicesyslog works on iOS > 13